src/array_list.c

```diff
@@ -172,7 +172,7 @@ rcutils_array_list_remove(rcutils_array_list_t * array_list, size_t index)
   if (copy_count > 0) {
     uint8_t * dst_ptr = rcutils_array_list_get_pointer_for_index(array_list, index);
     uint8_t * src_ptr = rcutils_array_list_get_pointer_for_index(array_list, index + 1);
-    memcpy(dst_ptr, src_ptr, array_list->impl->data_size * copy_count);
+    memmove(dst_ptr, src_ptr, array_list->impl->data_size * copy_count);
```

Comment:
I think it is worthwhile to add a comment here saying why we are using a `memmove` instead of a `memcpy`.

Reply:
Added a comment, let me know if it does a good enough job explaining the use or if you'd like it rephrased in any way?